File: PE35/PE35.py
import math

def isPrime(n):
    if (n < 2):
        return False
    if (n < 4):
        return True
    if (n == 4):
        return False
    if (n == 5):
        return True
    for i in range(2, int(n**0.5)+1):
        if (n % i == 0):
            return False
    return True

def shiftleft(sn):
    shifted = sn[1:]
    shifted += sn[0]
    return shifted

def checkcirc(sn):
    if (not isPrime(int(sn))):
        return False
    sn2 = shiftleft(sn)
    while (sn2 != sn):
        if (not isPrime(int(sn2))):
            return False
        sn2 = shiftleft(sn2)
    return True

count = 0
for i in range(2, 1000000):
    if (checkcirc(str(i))):
        count += 1
print(count)

File: PE35/PE35.java
package projecteuler35;

public class PE35 {
	static boolean isPrime(int n) {
		if (n < 2) return false;
		if (n < 4) return true;
		if (n == 4) return false;
		if (n == 5) return true;
		for (int i = 2; i < (int)Math.pow(n, 0.5)+1; i++) {
			if (n % i == 0) return false;
		}
		return true;
	}
	
	static String shiftleft(String sn) {
		String shifted = sn.substring(1);
		shifted += sn.charAt(0);
		return shifted;
	}
	
	static boolean checkcirc(String sn) {
		if (!isPrime(Integer.parseInt(sn))) return false;
		String sn2 = shiftleft(sn);
		while (!sn2.equals(sn)) {
			if (!isPrime(Integer.parseInt(sn2))) return false;
			sn2 = shiftleft(sn2);
		}
		return true;
	}
	
	public static void main(String args[]) {
		int count = 0;
		for (int i = 2; i < 1000000; i++) {
			if (checkcirc(Integer.toString(i))) count++;
		}
		System.out.println(count);
	}
}
